x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx

```diff
@@ -87,16 +118,32 @@ export const MainTabs: FC<Props> = ({ tabId, disableLinks }) => {
   }
 
   const tabs = getTabs(disableLinks);
+  const mlUrlGenerator = useMlUrlGenerator();
+  const navigateToPath = useNavigateToPath();
+
+  const redirectToTab = async (defaultPathId: MlUrlGeneratorState['page']) => {
+    // TODO: fix ts
+    // @ts-ignore
+    const path = await mlUrlGenerator.createUrl({
+      page: defaultPathId,
+      pageState: { globalState },
+    });
+
+    await navigateToPath(path, false);
+  };
+
+  useEffect(() => {
+    document.title = `ML - ${TAB_DATA[selectedTabId].name} - Kibana`;
+  }, [selectedTabId]);
 
   return (
     <EuiTabs display="condensed">
       {tabs.map((tab: Tab) => {
         const { id, disabled } = tab;
         const testSubject = TAB_DATA[id].testSubject;
-        const defaultPathId = TAB_DATA[id].pathId || id;
+        const defaultPathId = (TAB_DATA[id].pathId || id) as MlUrlGeneratorState['page'];
         // globalState (e.g. selected jobs and time range) should be retained when changing pages.
         // appState will not be considered.
-        const fullGlobalStateString = globalState !== undefined ? `?_g=${encode(globalState)}` : '';
```

Comment:
Why is the global state no longer necessary?

Reply:
I'm still retaining the globalState here, just within `redirectToTab` instead of the encoded string.